tests/phpunit/unit/api/compiler/beans-page-compiler/compilePageScripts.php

```diff
@@ -0,0 +1,387 @@
+		$assets                      = $this->get_mock_assets( [ 'admin-bar', 'uikit' ] );
+		$assets->registered['uikit'] = null;
+
+		// Set up the order of how beans_get() will be called.
+		Monkey\Functions\expect( 'beans_get' )
+			->ordered()
+			->once()
+			->with( 'wp_scripts', $GLOBALS )
+			->andReturn( $assets )
+			->andAlsoExpectIt()
+			->ordered()
+			->once()
+			->with( 'uikit', $assets->registered )
+			->andReturn( $assets->registered['uikit'] );
+
+		// Check that beans_compile_js_fragments() does not get called.
+		Monkey\Functions\expect( 'beans_compile_js_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_scripts() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_scripts() should not compile when the script has no src.
+	 */
+	public function test_should_not_compile_when_script_has_no_src() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\when( 'get_option' )->justReturn( true );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Initialize the mocked assets.
+		$assets                            = $this->get_mock_assets( [ 'admin-bar', 'uikit' ] );
+		$assets->registered['uikit']->src  = '';
+		$assets->registered['uikit']->deps = [];
+
+		// Set up the order of how beans_get() will be called.
+		Monkey\Functions\expect( 'beans_get' )
+			->ordered()
+			->once()
+			->with( 'wp_scripts', $GLOBALS )
+			->andReturn( $assets )
+			->andAlsoExpectIt()
+			->ordered()
+			->once()
+			->with( 'uikit', $assets->registered )
+			->andReturn( $assets->registered['uikit'] );
+
+		// Check that beans_compile_js_fragments() does not get called.
+		Monkey\Functions\expect( 'beans_compile_js_fragments' )->never();
+
+		// Run the tests.
+		$this->assertNull( ( new _Beans_Page_Compiler() )->compile_page_scripts() );
+	}
+
+	/**
+	 * Test _Beans_Page_Compiler::compile_page_scripts() should compile when the script has src but no dependencies.
+	 */
+	public function test_should_compile_when_script_has_src_but_no_deps() {
+		Monkey\Functions\when( 'beans_get_component_support' )->justReturn( true );
+		Monkey\Functions\expect( 'get_option' )
+			->ordered()
+			->once()
+			->with( 'beans_compile_all_scripts', false )
+			->andReturn( true )
+			->andAlsoExpectIt()
+			->ordered()
+			->once()
+			->with( 'beans_compile_all_scripts_mode', 'aggressive' )
+			->andReturn( 'aggressive' );
+		Monkey\Functions\when( '_beans_is_compiler_dev_mode' )->justReturn( false );
+		Monkey\Functions\expect( 'add_query_arg' )->never();
+
+		// Initialize the mocked assets.
+		$assets                                = $this->get_mock_assets( [ 'admin-bar', 'uikit', 'debug-bar' ] );
+		$assets->registered['uikit']->deps     = [];
+		$assets->registered['debug-bar']->deps = [];
+
+		// Set up the order of how beans_get() will be called.
```

Comment:
Maybe `Check the order`? 
or 
`Set up the check for the order of how ...`

`Set up the order` sounds like we could actually "influence" the order.